src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/utils/numeric_methods.py

```diff
@@ -15,13 +15,15 @@ def find_root(
     relative_convergence_tolerance: float = CONVERGENCE_TOLERANCE,
     maximum_number_of_iterations: int = MAXIMUM_NUMBER_OF_ITERATIONS,
 ) -> float:
-    """Root finding using scipy and the Brent's method.
+    """Root finding using scipy´s implementation of the brenth method.
+
+    TODO: Investigate why we don't use brentq method reccomended by scipy
```

Comment:
We use brenth because it is quite a bit faster than brentq:

https://dl.acm.org/doi/10.1145/355656.355659

Neither of the variants are guaranteed to converge under the conditions we are using, so we have a fallback on a simpler secant method.

The motivation to change to Brent's method was to speed things up.

Reply:
The paper is linked from: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.brenth.html#r044260d935ec-busanddekker1975